lumicks/pylake/piezo_tracking/baseline.py

```diff
@@ -1,6 +1,90 @@
-import numpy as np
 import matplotlib.pyplot as plt
+import numpy as np
 from lumicks.pylake.channel import Slice, Continuous
+from csaps import csaps
+from sklearn.model_selection import RepeatedKFold
+from sklearn.metrics import mean_squared_error
+
+
+def unique_sorted(trap_position, force):
+    """Sort and remove duplicates trap_position data to prepare for fit smoothing spline.
+    Parameters
+    ----------
+    trap_position : lumicks.pylake.Slice
+        Trap mirror position
+    force : lumicks.pylake.Slice
+        Force data
+
+    Returns
+    --------
+    x : np.array
+        sorted trap position data
+    force : np.array
+        sorted force data"""
+
+    x = trap_position.data
+    u, c = np.unique(x, return_counts=True)
+    m = np.isin(x, [u[c < 2]])
+    ind = np.argsort(x[m])
+
+    return x[m][ind], force.data[m][ind]
+
+
+def optimize_smoothing_factor(
+    trap_position,
+    force,
+    smoothing_factors,
+    n_repeats,
+    plot_smoothingfactor_mse,
```

Comment:
Considering that the function has `smoothing_factor` in the name, I reckon `plot_mse` would be sufficient.